woocommerce/woocommerce-gutenberg-products-block
# Playwright E2E tests: Multiple signed in roles

This PR moves the following tests to playwright and removes the puppeteer equivalent:

- Guest user can place order
- Logged in user can place an order

**Setting up Playwright's multiple sign up roles**

We are using [Multiple signed in roles](https://playwright.dev/docs/auth#multiple-signed-in-roles) to test for authenticated vs. unauthenticated users. There are some required adjustments to fix https://github.com/woocommerce/woocommerce-blocks/pull/10561#issuecomment-1691870998. We will update this section when the PR is ready to be merged!

- [x] Implement login/logout util function
- [x] Setup Playwright's multiple sign up roles

#### Other Checks


### Testing

#### Automated Tests

#### User Facing Testing

Ensure Playwright E2E tests pass and ensure Puppeteer E2Es also pass, so that we can be sure the removed tests don't impact on anything.

##### Test the multiple sign-in roles feature

1. In your local Woo Blocks plugin, set breakpoints in the test file `tests/e2e/tests/checkout/checkout-block.block_theme.side_effects.spec.ts` at the lines `39` and `114`
2. Run the test file `tests/e2e/tests/checkout/checkout-block.block_theme.side_effects.spec.ts` in debug mode
3. The first time the code is paused, check the Playwright's browser. Ensure you are on the shop page
4. Open the `My Account` page in a new tab. Ensure the customer `Jane Smith` is logged in
5. The second time the code is paused, check the Playwright's browser. Ensure you are on the shop page
6. Open the `My Account` page in a new tab. Ensure the user is logged out


### WooCommerce Visibility
